app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ReportingRepository.kt

```diff
@@ -170,37 +171,40 @@ class ReportingRepositoryImpl(
                 ?: throw InvalidConfigurationException.ConfigurationNotPresent
             val uploadKeysDays = configuration.uploadKeysDays
                 ?: throw InvalidConfigurationException.NullNumberOfDaysToUpload
-            val thresholdTimeFromConfiguration = ZonedDateTime.now()
+            val uploadStartIntervalNumberFromConfig = ZonedDateTime.now()
                 .minusDays(uploadKeysDays.toLong())
-                .startOfTheUtcDay()
                 .toRollingStartIntervalNumber()
 
             val dateWithMissingExposureKeys = dateWithMissingExposureKeys
 
             if (dateWithMissingExposureKeys != null) {
                 // Report only the exposure keys that have not been uploaded in the day of the previous submission.
                 val keysToUpload = teks.filter {
-                    it.rollingStartIntervalNumber >= dateWithMissingExposureKeys.startOfTheUtcDay().toRollingStartIntervalNumber() &&
-                        it.rollingStartIntervalNumber <= dateWithMissingExposureKeys.endOfTheUtcDay().toRollingStartIntervalNumber()
+                    it.rollingStartIntervalNumber >= dateWithMissingExposureKeys.toRollingStartIntervalNumber() &&
+                            it.rollingStartIntervalNumber <= dateWithMissingExposureKeys.endOfTheUtcDay()
+                        .toRollingIntervalNumber()
                 }
 
                 uploadTeksWithMessageType(keysToUpload, infectionLevel)
 
                 quarantineRepository.markMissingExposureKeysAsUploaded()
             } else {
                 // The regular flow of reporting the exposure keys.
-                val thresholdTime = if (infectionLevel == MessageType.InfectionLevel.Red) {
-                    diagnosisKeysRepository.getSentTeksByMessageType(
-                        MessageType.InfectionLevel.Yellow
-                    ).minBy { tekMetadata ->
-                        tekMetadata.rollingStartIntervalNumber
-                    }?.rollingStartIntervalNumber ?: thresholdTimeFromConfiguration
-                } else {
-                    thresholdTimeFromConfiguration
-                }
+                val uploadStartIntervalNumber =
+                    if (infectionLevel == MessageType.InfectionLevel.Red) {
+                        // If we sent a yellow warning before, send red warnings from the day of the
+                        // earliest yellow warning
+                        diagnosisKeysRepository.getSentTeksByMessageType(
```

Comment:
A question about the temporary `rollingPeriod` which is provided in the day of the reporting.

Given the following case:
- We report yellow with for rollingStartPeriod1/rollingPeriod1 ... rollingStartPeriod3/rollingPeriod3 and rollingStartPeriodToday, rollingPeriodToday (temporary) from the SDK
- Two days later days later we report red with rollingStartPeriod1/rollingPeriod1 ... rollingStartPeriod6/rollingPeriod6   + rollingStartPeriodToday, rollingPeriodToday

When reporting red for day 4 we have the final rollingStartPeriod/rollingPeriod which is different from the temporary one we had when reporting yellow. 

Do we need to send to the server TEKs with updated state for the temporary rollingStartPeriod/rollingPeriod?

Reply:
We get both. The TEK with the shortened rollingPeriod and a new one with 144.
Currently we would also upload the new 144-period TEK as red as well.